src/platform/backends/lxd/lxd_virtual_machine_factory.cpp

```diff
@@ -109,6 +76,39 @@ void mp::LXDVirtualMachineFactory::hypervisor_health_check()
                  fmt::format("{}: {}", base_url.toString(), QJsonDocument(reply).toJson(QJsonDocument::Compact)));
         throw std::runtime_error("Failed to authenticate to LXD.");
     }
+
+    try
+    {
+        lxd_request(manager.get(), "GET",
+                    QUrl(QString("%1/projects/%2").arg(base_url.toString()).arg(lxd_project_name)));
+    }
+    catch (const LXDNotFoundException&)
+    {
+        QJsonObject project{{"name", lxd_project_name}, {"description", "Project for Multipass instances"}};
+
+        lxd_request(manager.get(), "POST", QUrl(QString("%1/projects").arg(base_url.toString())), project);
+
+        // TODO: Detect if default storage pool is available and if not, create a directory based pool for
+        //       Multipass
+
+        QJsonObject devices{
+            {"eth0", QJsonObject{{"name", "eth0"}, {"nictype", "bridged"}, {"parent", "mpbr0"}, {"type", "nic"}}}};
+        QJsonObject profile{{"description", "Default profile for Multipass project"}, {"devices", devices}};
+
+        lxd_request(manager.get(), "PUT", QUrl(QString("%1/profiles/default").arg(base_url.toString())), profile);
+    }
+
+    try
+    {
+        lxd_request(manager.get(), "GET",
+                    QUrl(QString("%1/networks/%2").arg(base_url.toString()).arg(multipass_bridge_name)));
+    }
+    catch (const LXDNotFoundException&)
+    {
+        QJsonObject network{{"name", multipass_bridge_name}, {"description", "Network bridge for Multipass"}};
+
+        lxd_request(manager.get(), "POST", QUrl(QString("%1/networks").arg(base_url.toString())), network);
+    }
```

Comment:
This feels weird here… Should we instead have a `VirtualMachineFactory::initialize` that would only be called after the health checks pass?

Reply:
I believe making sure the project and the network bridge exist during the health check is exactly what the health check is for.  My assertion is what happens if a user deletes the project or network while multipass is running?  We'd want to check for that and create those if missing, right?  Otherwise, we can just pass an error up to the user to either recreate whatever is missing or tell them to restart `multipassd` and it will make sure they are there on start up.